Added a basic gym state wrapper

DavidSlayback opened the pull request:
1) Added GymStateObservationWrapper, a wrapper to replace observations with state. This wrapper leaves the underling environment untouched, so render and properties still work properly

2) Updated the scripts to add a state representation to the environments. Additionally updated the gym script; gym.make will throw a standard error if the provided id isn't a balid gym ID, not an UnregisteredEnv

3) Updated .gitignore to ignore PyCharm .idea directory

abaisero commented on scripts/gv_control_loop_gym.py:
```diff
@@ -11,19 +11,23 @@
         env = gym.make(id_or_path)
 
-    except gym.error.UnregisteredEnv:
+    except gym.error.Error:
```
Any reason why this error became more generic?  Did you encounter some errors which were not covered by UnregisteredEnv?

DavidSlayback replied:
As I mentioned in the original pull request, the issue is that gym won't throw an UnregisteredEnv Error if the environment id is "malformed", it'll just throw a standard Error. In realistic use cases where you would pass some yaml filepath to the function, the filepath won't be a valid gym id (valid being of the form ^(?:[\w:-]+\/)?([\w:.-]+)-v(\d+)$), so gym won't even check if such an environment is registered

abaisero replied:
Oh, I see, sorry, I missed that part.  It's annoying that we don't have a good system in place to test scripts;  I guess that's why that issue was not caught before.

abaisero commented on gym_gridverse/gym_wrappers.py:
```diff
@@ -0,0 +1,21 @@
+import gym
+
+class GymStateObservationWrapper(gym.Wrapper):
```
I think this should be called GymStateWrapper;  otherwise StateObservation is not very clear from the name alone.  There's also other functionality I'd like this to have, but that can be added as a future commit.

abaisero reviewed (changes requested):
Two more comments (others below):
* I think we'd rather have the GymState(Observation)Wrapper in the gym_gridverse/gym.py module, just under the GymEnvironment definition.  I don't see a reason to split them since they're both very simple implementations.
* I'd like to see a couple of simple tests in tests/test_gym.py, nothing fancy, just copy the tests for GymEnvironment, but using GymStateWrapper.

DavidSlayback commented:
Sorry for the delay, I've been running some experiments in other environments. I updated the request with your suggestions and commented on why I changed the Error type for the make_env function.
